src/adapter/src/rbac.rs

```diff
@@ -0,0 +1,236 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::catalog::Catalog;
+use crate::command::Command;
+use crate::session::Session;
+use crate::AdapterError;
+use mz_sql::catalog::{CatalogRole, SessionCatalog};
+use mz_sql::plan::Plan;
+use std::error::Error;
+use std::fmt;
+use std::fmt::Formatter;
+
+/// Attributes that allow a role to execute certain plans.
+///
+/// Note: This is a subset of all role attributes used for privilege checks.
+#[derive(Debug)]
+pub enum Attribute {
```

Comment:
Would it ever be possible for a role to have multiple attributes? If so, instead of representing `Attribute`s as an enum (and a group of `Attribute`s as a `Vec<Attribute>`?) what do you think about using [`bitflags`](https://docs.rs/bitflags/latest/bitflags/)?

Reply:
I think it would make sense to convert `RoleAttributes` to a bitflag, and then re-user that bitflag here. 
https://github.com/MaterializeInc/materialize/blob/534cdc4dbbab4a3a6a6067c2d2fc49ed9823026d/src/sql/src/catalog.rs#L284-L297
However, until https://github.com/MaterializeInc/materialize/issues/17824 is complete it's going to be hard to write that migration. I left a TODO above `RoleAttributes` with this info.

Reply:
Sounds good to me!